dataland-frontend/src/components/forms/CreateSFDRDataset.vue

```diff
@@ -0,0 +1,330 @@
+import SubmitSideBar from "@/components/forms/parts/SubmitSideBar.vue";
+import YesNoNaFormField from "@/components/forms/parts/fields/YesNoNaFormField.vue";
+import UploadReports from "@/components/forms/parts/UploadReports.vue";
+import DataPointFormField from "@/components/forms/parts/kpiSelection/DataPointFormField.vue";
+import PercentageFormField from "@/components/forms/parts/fields/PercentageFormField.vue";
+import ProductionSitesFormField from "@/components/forms/parts/fields/ProductionSitesFormField.vue";
+import { objectDropNull, ObjectType } from "@/utils/UpdateObjectUtils";
+import { smoothScroll } from "@/utils/SmoothScroll";
+import { DocumentToUpload, ReportToUpload, uploadFiles } from "@/utils/FileUploadUtils";
+import MostImportantProductsFormField from "@/components/forms/parts/fields/MostImportantProductsFormField.vue";
+import { Subcategory } from "@/utils/GenericFrameworkTypes";
+import ProcurementCategoriesFormField from "@/components/forms/parts/fields/ProcurementCategoriesFormField.vue";
+
+export default defineComponent({
+  setup() {
+    return {
+      getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
+    };
+  },
+  name: "CreateSfdrDataset",
+  components: {
+    SubmitButton,
+    SubmitSideBar,
+    UploadFormHeader,
+    SuccessMessage,
+    FailMessage,
+    FormKit,
+    Card,
+    PrimeButton,
+    Calendar,
+    YesNoFormField,
+    InputTextFormField,
+    FreeTextFormField,
+    NumberFormField,
+    DataPointFormField,
+    DateFormField,
+    SingleSelectFormField,
+    MultiSelectFormField,
+    NaceCodeFormField,
+    AddressFormField,
+    RadioButtonsFormField,
+    YesNoNaFormField,
+    PercentageFormField,
+    ProductionSitesFormField,
+    MostImportantProductsFormField,
+    ProcurementCategoriesFormField,
+    UploadReports,
+  },
+  directives: {
+    tooltip: Tooltip,
+  },
+  emits: ["datasetCreated"],
+  data() {
+    return {
+      formId: "createSFDRForm",
+      waitingForData: true,
+      dataDate: undefined as Date | undefined,
+      companyAssociatedSfdrData: {} as CompanyAssociatedDataSfdrData,
+      sfdrDataModel: sfdrDataModel,
+      route: useRoute(),
+      message: "",
+      smoothScroll: smoothScroll,
+      uploadSucceded: false,
+      postSfdrDataProcessed: false,
+      messageCounter: 0,
+      checkCustomInputs,
+      documents: [] as DocumentToUpload[],
+      referencedReportsForPrefill: {},
+      namesOfAllCompanyReportsForTheDataset: [] as string[],
+    };
+  },
+  computed: {
+    yearOfDataDate: {
+      get(): string {
+        const currentDate = this.companyAssociatedSfdrData.data?.general?.general?.fiscalYearEnd;
+        if (currentDate === undefined) {
+          return "";
+        } else {
+          const currentDateSegments = currentDate.split("-");
+          return currentDateSegments[0] ?? new Date().getFullYear();
+        }
+      },
+      set() {
+        // IGNORED
+      },
```

Comment:
You could also write this without the getter and setter syntax just like you have done it for the computed property `subcategoryVisibility` right?
Would save some lines of code if it was possible.

Reply:
I tried, but this is being used inside a v-model which expects a valid JavaScript member expression. We have similar code in CreateLksgDataset.vue

Reply:
Ok but in my opinion, it does not make sense to use a v-model here.
The `yearOfDataDate` should only change, if the underlying data `companyAssociatedSfdrData` changes.
It should never be the opposite way.
So why not chang ingit to a v-bind instead of a v-model?
Or does FormKit have problems with that? (I don't expect it to have problems with it).

I tried making it a v-bind on my machine locally and I also wrote the computed property like this;

![image](https://github.com/d-fine/Dataland/assets/99255296/1450c9de-c448-4a13-845f-ce969bef5312)

And it seemed work at first glance.

Same thing could then be done in `CreateLksgDataset.vue`.

Reply:
That's the code I tried earlier, but with v-model
v-bind seems to be fine in IDE and during compilation, but fails at runtime:

![image](https://github.com/d-fine/Dataland/assets/131248192/a973adf1-fe84-47bd-96b5-56adf287e642)

Reply:
Ahh ok so the computed property is read-only if you use the syntax I proposed.
Ok then let's stick to the way with getter and setter.